### ISSUE TITLE
DropdownToggleCheckbox causes React controlled-state warning when moving in or out of indeterminate state

### ISSUE THREAD

**Issue by mturley** [patternfly/patternfly-react#4213]
**Describe the issue. What is the expected and unexpected behavior?**

Our Checkbox and DropdownToggleCheckbox components both support passing a null value for the `isChecked` prop to display an indeterminate state (line through the box, not checked or unchecked). For the DropdownToggleCheckbox (and not the Checkbox), when that prop changes from false to null, React throws this warning:

> A component is changing a controlled input of type checkbox to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component.

Comparing the source for Checkbox and DropdownToggleCheckbox, the difference appears to be [this line](https://github.com/patternfly/patternfly-react/blob/master/packages/react-core/src/components/Checkbox/Checkbox.tsx#L74):
```ts
checkedProps.checked = checkedProps.checked === null ? false : checkedProps.checked;
```

So, we need the `checked` attribute on the underlying `<input>` to always be a boolean even when `isChecked` is null, since the null value just drives the `indeterminate` property via a ref callback.

I think the real fix for this is just to remove the native `<input>` from DropdownToggleCheckbox and instead reuse Checkbox in there (I think the dropdown one predated the generic one).

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

Use this CodeSandbox (thanks @redallen): https://codesandbox.io/s/dazzling-hill-wzf1j
If you uncheck all the child checkboxes, you'll see the parent checkbox switch from indeterminate to unchecked, and the error will appear in the console.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Bug. It causes a silent warning, but as long as the behavior still works as intended it's not a blocker.

**What is your product and what release version are you targeting?**
This is for @AjayJagan in Kogito. This is the PR where he is seeing the issue: https://github.com/kiegroup/kogito-apps/pull/209. Not sure what release version he is targeting.

**Comment by cristianonicolai** [comment]
@mturley Kogito currently uses this set of versions: https://github.com/kiegroup/kogito-apps/blob/master/packages/management-console/package.json#L39 from 2020.04 release. Thanks

**Comment by AjayJagan** [comment]
Hey Cristiano , Yes I think its ok to merge the PR.

**Comment by mturley** [comment]
This was fixed by #4844
